compiler/rustc_infer/src/infer/opaque_types.rs

```diff
@@ -470,7 +470,7 @@ where
         &mut self,
         t: &ty::Binder<'tcx, T>,
     ) -> ControlFlow<Self::BreakTy> {
-        t.as_ref().skip_binder().visit_with(self);
```

Comment:
This change is a bit weird to me. I don't think this is the same semantics, even if it might be the same behavior.